cypress/integration/resources.spec.js

```diff
@@ -239,4 +245,63 @@ describe('resources', () => {
   //   checkAllCardsForDataAttribute(DATA_TEST_LANGUAGES, 'JavaScript');
   //   checkAllCardsForDataAttribute(DATA_TEST_LANGUAGES, 'Python');
   // });
+  it('will prompt user to login when voting without being logged in', () => {
+    cy.server();
+    cy.route('POST', 'auth/login/').as('postLogin');
+
+    cy.findAllByTestId(UPVOTE_BUTTON).first().click({ force: true });
+    cy.findByTestId(LOGIN_FORM).should('exist');
+
+    cy.findByLabelText('Email*').type(existingUser.email);
+    cy.findByLabelText('Password*').type(existingUser.password);
+    cy.findByTestId(LOGIN_BUTTON).click();
+
+    cy.wait('@postLogin').its('status').should('eq', 200);
+
+    cy.findByTestId(LOGIN_FORM).should('not.exist');
+  });
+
+  it('will allow user to upvote once logged in', () => {
+    cy.findAllByTestId(UPVOTE_BUTTON)
+      .first()
+      .children('span')
+      .invoke('text')
+      .as('currentUpVoteCountText');
+
+    cy.server();
+    cy.route('PUT', 'api/v1/resources/**/upvote').as('upvote');
+
+    cy.findAllByTestId(UPVOTE_BUTTON).first().click({ force: true });
+    cy.wait('@upvote');
+
+    cy.get('@upvote').then(({ status, response }) => {
+      expect(status).to.eq(200);
+      cy.get('@currentUpVoteCountText').should(
+        'not.eq',
+        `Number of Number of upvotes:${response.body.resource.upvotes}`,
+      );
+    });
+  });
```

Comment:
I feel like these are a bit mixed.

`> will prompt user to login when voting without being logged in` but then you assert that the login form works

I would consider getting rid of the login here, remove the preserved cookie, and simply make the act of logging in and then voting one test